x-pack/plugins/security_solution/common/endpoint/generate_data.ts

```diff
@@ -450,6 +556,7 @@ export class EndpointDocGenerator {
   ): AlertEvent {
     return {
       ...this.commonInfo,
+      ...this.alertsDataStream,
```

Comment:
But you know the stream why don't you call a method that take parameters here and returns the object with the stream information. And I am not too clear on the seeding changes.

Reply:
We could do that but we'd have to pass it through quite a few function calls when building a full tree instead of just a single alert. I think it's simpler to set it in the constructor and make it part of the object.

Reply:
actually... I think that's a better idea, I'll switch it.